examples/amm/Cargo.toml

```diff
@@ -0,0 +1,35 @@
+[package]
+name = "amm"
+version = "4.0.0-alpha.1"
+authors = ["Parity Technologies <admin@parity.io>"]
+edition = "2021"
+publish = false
+
+[dependencies]
+ink = { path = "../../crates/ink", default-features = false }
+
+scale = { package = "parity-scale-codec", version = "3", default-features = false, features = ["derive"] }
+scale-info = { version = "2", default-features = false, features = ["derive"], optional = true }
+
+[lib]
+name = "amm"
+path = "lib.rs"
+crate-type = [
+	"cdylib",
+]
+
+# This is necessary to get things to compile. Without it the compiler throws an error.
+# Setting overflow checks to "false" for an AMM is typically not a good idea...
+# "This typically goes back to a known bug in the Rust compiler:
+# https://github.com/rust-lang/rust/issues/78744"
+[profile.release.package.compiler_builtins]
+overflow-checks = false
```

Comment:
You should be able to get rid of this now too, I imagine it part of your problems stemmed from being on the `3.x` series of releases